Examples/Io/Obj/src/ObjSimHitWriter.cpp

```diff
@@ -34,22 +26,37 @@
 
 namespace {
 
+/// @brief Helper function to interpolate points
+///
+/// @tparam input_vector_type
+/// @param inputs input vector points
+/// @param nPoints number of interpolation points
+///
+/// @return std::vector<Acts::Vector3> interpolated points
 template <typename input_vector_type>
 std::vector<Acts::Vector3> interpolatedPoints(
-    const std::vector<input_vector_type>& inputs, unsigned int nPoints) {
-  Eigen::MatrixXd points(3, inputs.size());
-  for (unsigned int i = 0; i < inputs.size(); ++i) {
-    points.col(i) = inputs[i].template head<3>().transpose();
-  }
-  Eigen::Spline<double, 3> spline3D =
-      Eigen::SplineFitting<Eigen::Spline<double, 3>>::Interpolate(points, 2);
-
+    const std::vector<input_vector_type>& inputs, std::size_t nPoints) {
   std::vector<Acts::Vector3> output;
-  double step = 1. / (nPoints - 1);
-  for (unsigned int i = 0; i < nPoints; ++i) {
-    double t = i * step;
-    Eigen::Vector3d point = spline3D(t);
-    output.push_back(Acts::Vector3(point[0], point[1], point[2]));
+
+  if (nPoints < 1) {
```

Comment:
```suggestion
  if (nPoints < 2) {
```
otherwise we run into problems at `double step = 1. / (nPoints - 1);`